test/Attack.ts

```diff
@@ -0,0 +1,212 @@
+import chai, { expect } from "chai";
```

Comment:
If we fix the bug, these tests will fail right? Should we repurpose this test, and put it in `OriginationController.ts`, also making sure that we check for failure via a revert as opposed to the test passing?

Reply:
This test will not fail after fixing the bug because of its socially engineered flow:

- the signature is [created](https://github.com/arcadexyz/v2-contracts/blob/980d793682fee9e9a139b54d94b0bca1a789ae35/test/Attack.ts#L178) by the lender and specifies the borrower for `Side`
- the vault is initialized by the lender and is [transferred](https://github.com/arcadexyz/v2-contracts/blob/980d793682fee9e9a139b54d94b0bca1a789ae35/test/Attack.ts#L190) to the borrower
- borrower approves bundleId of aforementioned vault to originationController
- borrower gives [approve](https://github.com/arcadexyz/v2-contracts/blob/980d793682fee9e9a139b54d94b0bca1a789ae35/test/Attack.ts#L195) to lender to sign and initialize loans on their behalf -- hence the signature created by the lender, specifying the Borrow as side is OK because of this approve 
-  [initializeLoan](https://github.com/arcadexyz/v2-contracts/blob/980d793682fee9e9a139b54d94b0bca1a789ae35/test/Attack.ts#L205) is run and `_validateCounterparties` is called, where `Side` was set to `B` by the lender but the borrower sanctioned them to do so with the `approve`, so no `revert` occurs.

I think we should remove this test. It is too much of an edge case.
Do you agree?